cli/add/add.go

```diff
@@ -105,44 +109,56 @@ func HandleAdd(args []string) (int, error) {
 }
 
 // TODO(siggisim): Support specifying a version.
-func FetchModuleOrDisambiguate(module string) (string, *RegistryResponse, error) {
-	module = strings.Replace(module, "https://", "", 1)
-	module = strings.Replace(module, "github.com/", "github/", 1)
-	module = strings.TrimRight(module, "/")
-
-	res, err := fetch(module)
+func FetchModuleOrDisambiguate(moduleInput string) (string, string, *RegistryResponse, error) {
+	moduleAndVersion := strings.Replace(moduleInput, "https://", "", 1)
+	moduleAndVersion = strings.Replace(moduleAndVersion, "github.com/", "github/", 1)
+	moduleAndVersion = strings.TrimRight(moduleAndVersion, "/")
+	moduleParts := strings.Split(moduleAndVersion, "@")
+	moduleName := moduleParts[0]
+	moduleVersion := ""
+	if len(moduleParts) > 1 {
+		moduleVersion = moduleParts[1]
+	}
+	res, err := fetch(moduleName)
 	if err != nil {
-		return "", nil, err
+		return "", "", nil, err
 	}
 	if len(res.Disambiguation) == 0 && res.Name == "" {
-		return "", nil, fmt.Errorf("module %q not found", module)
+		return "", "", nil, fmt.Errorf("module %q not found", moduleName)
 	}
 	if len(res.Disambiguation) == 1 && res.Name == "" {
-		module = res.Disambiguation[0].Path
-		res, err = fetch(module)
+		moduleName = res.Disambiguation[0].Path
+		res, err = fetch(moduleName)
 		if err != nil {
-			return "", nil, err
+			return "", "", nil, err
 		}
 	}
 	if len(res.Disambiguation) > 1 && res.Name == "" {
 		pickedModule, err := showPicker(res.Disambiguation)
 		if err != nil {
-			return "", nil, err
+			return "", "", nil, err
 		}
-		module = pickedModule
-		res, err = fetch(module)
+		moduleName = pickedModule
+		res, err = fetch(moduleName)
 		if err != nil {
-			return "", nil, err
+			return "", "", nil, err
 		}
 	}
-	return module, res, nil
+	return moduleName, moduleVersion, res, nil
 }
 
-func GenerateSnippet(module string, resp *RegistryResponse) string {
-	versionKey := fmt.Sprintf("[https://registry.build/%s@%s]", module, resp.LatestReleaseWithWorkspaceSnippet)
+func GenerateSnippet(module, version string, resp *RegistryResponse) string {
+	versionKey := fmt.Sprintf("[https://registry.build/%s@%s]", module, version)
 	header := fmt.Sprintf(headerTemplate, versionKey)
 	footer := fmt.Sprintf(footerTemplate, versionKey)
-	return fmt.Sprintf("\n%s\n\n%+v\n\n%s\n", header, strings.TrimSpace(resp.WorkspaceSnippet), footer)
+	snippet := resp.WorkspaceSnippet
+	for _, r := range resp.Releases {
+		if r.Name == "v"+version || r.Name == version {
```

Comment:
👀

Reply:
Haha it's not strictly required, just adds some flexibility in specifying version names